src/requestor/routes/query.py

```diff
@@ -84,26 +97,39 @@ async def list_requests(
 @router.get("/request/user", status_code=HTTP_200_OK)
 async def list_user_requests(api_request: Request, auth=Depends(Auth)) -> dict:
     """
-    List the current user's requests.
+    List current user's requests.
 
-    Use the "active" query parameter to get only those requests
+    Use "active" query parameter to get only those requests
     created by the user that are not in DRAFT or FINAL statuses.
     """
     # no authz checks because we assume the current user can read
     # their own requests.
     active = False
-    if "active" in api_request.query_params:
-        if api_request.query_params["active"]:
+    filter_dict = {}
+    flat_params = flatten_query_params(api_request.query_params)
+    for param in flat_params:
```

Comment:
we're iterating over the parameters twice (in `flatten_query_params` and in this `for` loop), and storing them twice (by copying `flat_params` into `filter_dict`).

could we simplify to a single loop? you can check the logic [here](https://github.com/uc-cdis/audit-service/blob/ec29058/src/audit/routes/query.py#L107-L130) - by removing the extra start/stop/count/groupby logic we should end up with some simple logic

Reply:
I agree to that, I just thought flat params could be a function that can be generalized and reused. This approach felt more readable.

Reply:
you can make the change and if you want, still keep it as a function 🙂